### DISCUSSION TITLE
Feature request: Route scoped hooks

### DISCUSSION THREAD

**Discussion by joshnuss** [sveltejs/kit#12178] [Ideas]
A common use-case for hooks is scoping logic to a specific route and its descendants.

It might simplify things if `hooks.*` files could live inside the `src/routes` folder.
That would make it very easy to set up path-specific hooks.

For example:

```
src/routes
   +hooks.server.js
    /admin
       +hooks.server.js
    /api
       +hooks.server.js
```

**Comment by acoyfellow** [comment]
would `/admin/+hooks.server.js`:

- override `/+hooks.server.js`
- run before `/+hooks.server.js`
- run after `/+hooks.server.js`

**Reply by joshnuss** [reply]
It should not override, it would be nested.

For example, requesting `/admin`, would run two levels of hooks:

- global hooks `src/routes/+hooks.server.js`
- `/admin` hooks in `src/routes/admin/+hooks.server.js`

Hooks can run code before and after. It's up to the hook to call `resolve(event)` and place logic before and/or after.

**Comment by teemingc** [comment]
There’s an existing issue for this although it’s been buried https://github.com/sveltejs/kit/issues/6731
